src/transformers/modeling_tf_utils.py

```diff
@@ -601,6 +603,14 @@ def generate(
                 "Please use another model class (e.g. `TFOpenAIGPTLMHeadModel`, `TFXLNetLMHeadModel`, `TFGPT2LMHeadModel`, `TFCTRLLMHeadModel`, `TFT5ForConditionalGeneration`, `TFTransfoXLLMHeadModel`)"
             )
 
+        if self.config.is_encoder_decoder:
+            bos_token_id = bos_token_id if bos_token_id is not None else self.config.decoder_start_token_id
```

Comment:
I don't understand why we get rid of `decoder_start_token_id` but still use it here as `config.decoder_start_token_id`.
Also I'm not sure if this works in general. Bart e.g. has a different `decoder_start_token_id` than `bos_token_id`.

Also please always check the hard-coded integration tests of at least Bart, T5 and GPT2 with your changes:
`RUN_SLOW=1 pytest tests/test_modeling_bart.py`
`RUN_SLOW=1 pytest tests/test_modeling_gpt2.py`
`RUN_SLOW=1 pytest tests/test_modeling_t5.py`